feature/experimental/backup_nhg/ate_tests/backup_nhg_test/backup_nhg_test.go

```diff
@@ -206,9 +206,9 @@ func TestDirectBackupNexthopGroup(t *testing.T) {
 	for _, tc := range cases {
 		t.Run(tc.desc, func(t *testing.T) {
 			t.Run("Validate Baseline AFT Telemetry", func(t *testing.T) {
-				tcArgs.validateAftTelemetry(t, *deviations.DefaultNetworkInstance, nhip, atePort2.IPv4)
-				tcArgs.validateAftTelemetry(t, vrfA, dstPfx, nhip)
-				tcArgs.validateAftTelemetry(t, vrfB, dstPfx, atePort3.IPv4)
+				tcArgs.validateAftTelemetry(t, *deviations.DefaultNetworkInstance, nhip, atePort2.IPv4, atePort2.IPv4)
+				tcArgs.validateAftTelemetry(t, vrfA, dstPfx, nhip, atePort2.IPv4)
+				tcArgs.validateAftTelemetry(t, vrfB, dstPfx, atePort3.IPv4, atePort3.IPv4)
```

Comment:
Why do we need to extend this function? 
Since we are only checking aft during the stable state, why do we need to pass resolvedNhIpAddress ?

1: when we check aft for prefix nhip (192.0.2.254), we always expect NH to be Port2 ip (192.0.2.6)
2: when we check aft for prefix 198.51.100.0 under vrfA, it will always point to nhip address (192.0.2.254) 
3: when we check aft for prefix 198.51.100.0 under vrfB, we expect NH to be Port3 ip (192.0.2.10)

Reply:
1. when we check aft for prefix nhip (192.0.2.254), we always expect NH to be Port2 ip (192.0.2.6)
2: when we check aft for prefix 198.51.100.0 under vrfA, it will always point to nhip address (192.0.2.254) ----> In recursively resolved devices, this NH IP can be 192.0.2.6.  198.51.100.0--> 192.0.2.254 --> 192.0.2.6
3: when we check aft for prefix 198.51.100.0 under vrfB, we expect NH to be Port3 ip (192.0.2.10)

Since same function is being used, extended the function to provide nhip and resolved ip after recursive resolution.

Refer https://github.com/openconfig/featureprofiles/pull/850 for similar change in TE-3.1.

Reply:
I learned about the little behavioral difference between vendors for the aft return info, thanks for providing details for clarification.